### PR TITLE
[14.0][FW] [13.0] stock_dynamic_routing: fix compatibility with stock_move_source_relocate module

### PR DESCRIPTION
**Pull request by sebalix**
Port of #232 from 13.0 to 14.0.

### CODE REVIEW

File: stock_dynamic_routing/models/stock_move.py [hunk 1]
```diff
@@ -238,13 +239,27 @@ def _apply_routing_rule_pull(self, routing_details):
...
+        for move, routing_rule in routing_to_apply:
+            if routing_rule.method == "push":
+                # In this case, we should not assign the move inside the
+                # pull dynamic routing. The push move must first be added before the
+                # initial move is assigned. Otherwise, when the destination
+                # location of the initial move is changed by the push rule, the
+                # putaway won't be recomputed as it is already assigned.
+                continue
+
+            if not routing_rule:
```
Comment (hparfr) [inline stock_dynamic_routing/models/stock_move.py]: it makes more sense to but this condition before routing_rule.method == push.

File: stock_dynamic_routing/models/stock_move.py [hunk 2]
```diff
@@ -318,64 +351,55 @@ def _apply_routing_rule_pull(self, routing_details):
...
+                qty = sum(orig_moves.mapped("product_qty"))
+                split_move = move
+                split_move_vals = move._split(qty)
+                if split_move_vals:
+                    split_move = self.create(split_move_vals)
+                if split_move != move:
+                    # No split occurs if the quantity was the same.
+                    # But if it did split, detach it from the move on which
+                    # we have a different routing
+                    move.move_orig_ids -= orig_moves
```
Comment (hparfr) [inline stock_dynamic_routing/models/stock_move.py:401]: minor remark: it's ok for perfs to make a write in a for ?

**Review by hparfr — APPROVED** [review]
Minor remarks

File: stock_dynamic_routing/models/stock_move.py (hunk 2 above)
Comment (sebalix) [reply stock_dynamic_routing/models/stock_move.py:401]: We didn't spotted any issue regarding performance so far, but yes that could be improved maybe especially the use `|= orig_move` above to instead use IDs.
=> https://gist.github.com/guewen/a4a2eba5d128b4ed3129866f80571fbd

**Comment by sebalix** [comment]
Changes on `stock_dynamic_routing` broke the tests, it's not related to `stock_move_source_relocate_dynamic_routing` (I reproduce locally), checking...

**Comment by sebalix** [comment]
Fixed, I forgot to confirm the move created when splitting.

File: stock_dynamic_routing/models/stock_move.py (hunk 1 above)
Comment (sebalix) [reply stock_dynamic_routing/models/stock_move.py]: done (amended the current commit)

**Review by TDu — APPROVED** [review]

**Pull request merged**